### ISSUE TITLE
Walk Data Entry - Allow user to choose whether to display preferred or common names

### ISSUE THREAD

**Issue by andrewvanbreda** [BiologicalRecordsCentre/ABLE#530]
See discussion here
https://github.com/BiologicalRecordsCentre/ABLE/issues/529

Note to myself:
Store option is sample attribute (so it is remembered in edit mode).
Good starting point may be to look at the SPRING project which has a similar option for the user to choose whether to display butterflies or bees.

**Comment by CrisSevilleja** [comment]
Hi, I arrived late to the discussion in #529 but just to mention, I think it is important to allow the user to choose to see Latin or common names and not show by default both names at the same time. 
Several languages don't use common names at all (Spanish, Italian, Portuguese) and they are not interested on English common names. Thank you for solving this functionality.

**Comment by DavidRoy** [comment]
@andrewvanbreda are you able to work on this? Needed for Japan bms.

**Comment by andrewvanbreda** [comment]
@DavidRoy 
Yes, I had intended to work on it.
But I am juggling a lot of work, but it is mostly all BRC work at moment. I guess could you let me know how urgent this is, and I will try to juggle as best I can.

**Comment by DavidRoy** [comment]
@andrewvanbreda how much work is this to implement. The priority is the input forms. For editing, presumably the users can select the preferred option again?

**Comment by andrewvanbreda** [comment]
@DavidRoy 
I would store the option as a sample attribute.
So the option would be saved against the sample, so you would even be able to change the mode on an existing sample.
So editing would open up the page in the same mode as when you initially added it.

In theory it isn't super difficult.......but I hesitate to say that, because there are often little things that get in the way.
It isn't days and days, but it isn't 1 or 2 hours eithers. 
I may as well look now as it sounds like people are keen on this.

**Comment by DavidRoy** [comment]
@andrewvanbreda I don’t think it’s essential to store the choice as a sample attribute? The user can select the language choice when editing?

**Comment by andrewvanbreda** [comment]
Hi @DavidRoy
They would still be able to make the selection, the radio button would still be on the first page, it would just default to what the sample was saved as.
The advantage of an attribute would be that the user wouldn't have to make a different selection when editing unless they wanted.
I had envisioned it working much like SPRING's Butterfly/Hoverfly/Bee species list selector.
It isn't harder to code if it is an attribute by the way (perhaps even a little easier as the control is automatically generated)

The problem if it isn't an attribute, then the selection has to be made each time the page is entered.
(We could save a default in the page configuration, but every time the page opens it would always go back to the default, no matter what the user wants).

I guess the third possibility is perhaps to try and save the selection in a cookie, so the user makes a selection, but then it tries to remember what they like (providing they don't switch browser etc). That wouldn't be an attribute.
I am not overly familiar with cookies, but am sure I could look at other code to work out how this would work.

I guess a 4th option would be to have the default they want as an option on their user account page.

Let me know what you want.

**Comment by DavidRoy** [comment]
@andrewvanbreda we can keep this simple initially by implementing a radio button to switch. The default should be Latin. We can then revisit whether more complexity is needed?

**Comment by andrewvanbreda** [comment]
@DavidRoy 
OK I will do a non-attribute basic switcher.
I am not quite sure if that will need a page reload or not, might depend on how the grid works.

**Comment by andrewvanbreda** [comment]
Hi @DavidRoy @CrisSevilleja 
Th switcher is now available at the bottom of the first page (the sample page).
Unfortunately I had to abandon my initial attempt to put a switcher on the occurrence tab as there were lots of problems redrawing the grid, it was a lot more complex than I expected.

**Comment by CrisSevilleja** [comment]
Hi @andrewvanbreda that is exactly what we wanted, it is good and working. However, the place is not the best. I imagine you tried to place this switcher on the second page (species abundances page) where is more logical to select the display preferred. 
For now it is okay, the functionality is there but the place should change to the second page whenever possible. thanks

**Comment by andrewvanbreda** [comment]
Hi @CrisSevilleja
Yes this is what I first attempted to do.

There are basically two ways to approach this issue and create the switcher.
1. Don't put the switcher on the page itself, and then tthe page can know about the option as it first loads (so the code is running more on the server).
2. Or as a real-time switcher on the same page (where the code runs more inside your browser).

So what I have done is option 1, where the option is select before the species tab page is loaded.
As you say, option 2 is desirable and what you suggest, but I encountered problems doing it that way.
I hoped I would be able to just put the switcher onto the pagem and then call the existing code that redraws the grid (such as if the user changes the sort order), however this was very problematic.
Just one example of many odd behaviours I encountered was the system wouldn't even attempt to switch between preferred to common if the species had a number entered against it.

Leave it with me, leave the issue open, and I will try to fix this one day when I can. I cannot at the moment as I would need to change the code in much greater detail (I didn't write the original code).

Andy
